src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.ClearCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.InvalidCommand;
+import duke.command.ListCommand;
+import duke.command.ToggleCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+public class Parser {
+
+    /**
+     * Parses `String` to `Integer`.
+     *
+     * @param strToIntValue the `String` that needs to be parsed to an `Integer` value
+     * @return the `Integer` value of the `String`.
+     * @throws DukeException if the parameter could not be parsed to an `Integer`.
+     */
+    public static int parseInt(String strToIntValue) throws DukeException {
```

Comment:
Regarding this point my rationale was that I want the errors to be thrown as and where they happen when I was implementing the parts that needed `parseInt()` which was `DeleteCommand`, and `ToggleCommand`. 

In those commands I've also thrown in out-of-bounds errors, in those 2 files I've also done some "error-checking" as UI was available to me there and I was not keen on passing UI into the Parser (since parser should only really be processing strings)

Reply:
Ahh, I see, it makes more sense now. Thanks for the explanation!